### ISSUE TITLE
Error: The runs.using key in action.yml must be one of: [composite docker node12 node16], got node20

### ISSUE THREAD

**Issue by darklight147** [nektos/act#2050]
### Bug report info

```plain text
act version:            0.2.49
GOOS:                   darwin
GOARCH:                 arm64
NumCPU:                 8
Docker host:            DOCKER_HOST environment variable is not set
Sockets found:
        /var/run/docker.sock
        $HOME/.docker/run/docker.sock
Config files:           
        /Users/quasimodo/.actrc:
                -P ubuntu-latest=catthehacker/ubuntu:act-latest
                -P ubuntu-22.04=catthehacker/ubuntu:act-22.04
                -P ubuntu-20.04=catthehacker/ubuntu:act-20.04
                -P ubuntu-18.04=catthehacker/ubuntu:act-18.04
Build info:
        Go version:            go1.20.6
        Module path:           command-line-arguments
        Main version:          
        Main path:             
        Main checksum:         
        Build settings:
                -buildmode:           exe
                -compiler:            gc
                -ldflags:             -X main.version=0.2.49
                CGO_ENABLED:          1
                CGO_CFLAGS:           
                CGO_CPPFLAGS:         
                CGO_CXXFLAGS:         
                CGO_LDFLAGS:          
                GOARCH:               arm64
                GOOS:                 darwin
Docker Engine:
        Engine version:        24.0.6
        Engine runtime:        runc
        Cgroup version:        2
        Cgroup driver:         cgroupfs
        Storage driver:        overlay2
        Registry URI:          https://index.docker.io/v1/
        OS:                    Docker Desktop
        OS type:               linux
        OS version:            
        OS arch:               aarch64
        OS kernel:             6.4.16-linuxkit
        OS CPU:                8
        OS memory:             5939 MB
        Security options:
                name=seccomp,profile=unconfined
                name=cgroupns
```

### Command used with act

```sh
`act pull_request -j deployment -s GITHUB_TOKEN=$(gh auth token) -W .github/workflows/pr.yml`
```

### Describe issue

```bash
act pull_request -j deployment -s GITHUB_TOKEN=$(gh auth token) -W .github/workflows/pr.yml -P ubuntu-latest=node:20
WARN  ⚠ You are using Apple M-series chip and you have not specified container architecture, you might encounter issues while running act. If so, try running it with '--container-architecture linux/amd64'. ⚠  
[Preview PR/build] 🚀  Start image=node:20
[Preview PR/build]   🐳  docker pull image=node:20 platform= username= forcePull=true
[Preview PR/build] using DockerAuthConfig authentication for docker pull
[Preview PR/build]   🐳  docker create image=node:20 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[Preview PR/build]   🐳  docker run image=node:20 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[Preview PR/build]   ☁  git clone 'https://github.com/docker/setup-buildx-action' # ref=v3
[Preview PR/build] 🏁  Job succeeded
Error: The runs.using key in action.yml must be one of: [composite docker node12 node16], got node20
```

### Workflow content

```yml
# FILEPATH: /Users/quasimodo/Documents/workspace/pr-preview/portfolio-astro/.github/workflows/pr.yml

name: Preview PR

on:
  pull_request:
    branches: [ main ]
    types: [opened, synchronize, reopened]

  workflow_dispatch: 

permissions:
  contents: write
  pull-requests: write
  deployments: write
  actions: write
  checks: write
  statuses: write
  

concurrency: 
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

env:
  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
  AWS_REGION: ${{ secrets.AWS_REGION }}
  EKS_CLUSTER_NAME: pr-preview
  IS_FORK: ${{ github.event.pull_request.head.repo.full_name != github.repository }}

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3

      - name: Login to DockerHub
        uses: docker/login-action@v3
        with:
          username: ${{ env.DOCKER_USERNAME}}
          password: ${{ env.DOCKER_PASSWORD }}

      - name: Build and push Docker image
        uses: docker/build-push-action@v5
        with:
          context: .
          push: ${{ env.IS_FORK == 'false' }}
          tags: "quasimodo147/portfolio-astro:preview-pr-${{ github.event.pull_request.number }}-${{ github.event.pull_request.head.sha }}"
          cache-to: type=registry,ref=quasimodo147/portfolio-astro:cache-pr-${{ github.event.pull_request.number }}
          cache-from: type=registry,ref=quasimodo147/portfolio-astro:cache-pr-${{ github.event.pull_request.number }}

      - name: publish helm package
        if: ${{ env.IS_FORK == 'false' }}
        run: |
          helm package ./k8s/pr-preview --destination ./k8s/pr-preview/artifacts
          
      - name: publish pipeline artifact
        uses: actions/upload-artifact@v3
        if: ${{ env.IS_FORK == 'false' }}
        with:
          name: helm-package
          path: ./k8s/pr-preview/artifacts

  deployment:
    runs-on: ubuntu-latest
    environment:
      name: preview-pr-${{ github.event.pull_request.number }}
      url: https://pr${{ github.event.pull_request.number }}-${{ github.event.pull_request.head.sha }}-preview.flavio.software
    needs: build
    if: github.event.pull_request.head.repo.full_name == github.repository
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
        
      - name: Set up AWS CLI
        uses: aws-actions/configure-aws-credentials@v4
        with:
          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
          aws-region: ${{ env.AWS_REGION }}

      - name: Set up kubectl
        uses: azure/setup-kubectl@v3
        with:
          version: 'latest'

      - name: populate kubeconfig
        run: aws eks --region ${{ env.AWS_REGION }} update-kubeconfig --name ${{ env.EKS_CLUSTER_NAME }}

      - run: |
          branch_sha=$(echo "pr${{ github.event.pull_request.number }}-${{ github.event.pull_request.head.sha }}")
          namespace=$(echo "preview-pr-${{ github.event.pull_request.head.repo.name }}-${{ github.event.pull_request.number }}")
      
          echo "NAMESPACE=${namespace}" >> $GITHUB_ENV
          echo "BRANCH_SHA=${branch_sha}" >> $GITHUB_ENV
        name: Set environment variables
  
        # apply a helm chart
      - name: Setup Helm
        uses: azure/setup-helm@v3
        with:
          version: 'v3.2.4'
          
      - name: Install chart
        run: |
          helm upgrade \
          --install \
          preview-pr-${{ github.event.pull_request.number }} ./k8s/pr-preview \
          --set image.tag="preview-pr-${{ github.event.pull_request.number }}-${{ github.event.pull_request.head.sha }}" \
          --set branch=$BRANCH_SHA \
          -n $NAMESPACE \
          --create-namespace \
          --wait
```

### Relevant log output

```sh
onsistency: BindOptions:<nil> VolumeOptions:<nil> TmpfsOptions:<nil> ClusterOptions:<nil>}] MaskedPaths:[] ReadonlyPaths:[] Init:<nil>}
[Preview PR/build] [DEBUG] Created container name=act-Preview-PR-build-d5e167f82ffe4fa1c7de90ba0b9f53096c9c177f9614a4bd6cc1fd8a9f19b9c5 id=af91d489be091ec58aa912de0bcd31c1510619d0a1433ecb74feb66cd8d88bbe from image catthehacker/ubuntu:act-latest (platform: )
[Preview PR/build] [DEBUG] ENV ==> [RUNNER_TOOL_CACHE=/opt/hostedtoolcache RUNNER_OS=Linux RUNNER_ARCH=ARM64 RUNNER_TEMP=/tmp LANG=C.UTF-8]
[Preview PR/build]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[Preview PR/build] [DEBUG] Starting container: af91d489be091ec58aa912de0bcd31c1510619d0a1433ecb74feb66cd8d88bbe
[Preview PR/build] [DEBUG] Started container: af91d489be091ec58aa912de0bcd31c1510619d0a1433ecb74feb66cd8d88bbe
[Preview PR/build] [DEBUG] Writing entry to tarball workflow/event.json len:2
[Preview PR/build] [DEBUG] Writing entry to tarball workflow/envs.txt len:0
[Preview PR/build] [DEBUG] Extracting content to '/var/run/act/'
[Preview PR/build] [DEBUG] Found revision: d6a347bf25f4ad74dc59b213c0b9d502eb14eb4e
[Preview PR/build] [DEBUG] expression '${{ secrets.AWS_REGION }}' rewritten to 'format('{0}', secrets.AWS_REGION)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', secrets.AWS_REGION)'
[Preview PR/build] [DEBUG] expression 'format('{0}', secrets.AWS_REGION)' evaluated to '%!t(string=)'
[Preview PR/build] [DEBUG] expression '${{ secrets.AWS_SECRET_ACCESS_KEY }}' rewritten to 'format('{0}', secrets.AWS_SECRET_ACCESS_KEY)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', secrets.AWS_SECRET_ACCESS_KEY)'
[Preview PR/build] [DEBUG] expression 'format('{0}', secrets.AWS_SECRET_ACCESS_KEY)' evaluated to '%!t(string=)'
[Preview PR/build] [DEBUG] expression '${{ github.event.pull_request.head.repo.full_name != github.repository }}' rewritten to 'format('{0}', github.event.pull_request.head.repo.full_name != github.repository)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', github.event.pull_request.head.repo.full_name != github.repository)'
[Preview PR/build] [DEBUG] expression 'format('{0}', github.event.pull_request.head.repo.full_name != github.repository)' evaluated to '%!t(string=true)'
[Preview PR/build] [DEBUG] expression '${{ secrets.DOCKER_USERNAME }}' rewritten to 'format('{0}', secrets.DOCKER_USERNAME)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', secrets.DOCKER_USERNAME)'
[Preview PR/build] [DEBUG] expression 'format('{0}', secrets.DOCKER_USERNAME)' evaluated to '%!t(string=)'
[Preview PR/build] [DEBUG] expression '${{ secrets.DOCKER_PASSWORD }}' rewritten to 'format('{0}', secrets.DOCKER_PASSWORD)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', secrets.DOCKER_PASSWORD)'
[Preview PR/build] [DEBUG] expression 'format('{0}', secrets.DOCKER_PASSWORD)' evaluated to '%!t(string=)'
[Preview PR/build] [DEBUG] expression '${{ secrets.AWS_ACCESS_KEY_ID }}' rewritten to 'format('{0}', secrets.AWS_ACCESS_KEY_ID)'
[Preview PR/build] [DEBUG] evaluating expression 'format('{0}', secrets.AWS_ACCESS_KEY_ID)'
[Preview PR/build] [DEBUG] expression 'format('{0}', secrets.AWS_ACCESS_KEY_ID)' evaluated to '%!t(string=)'
[Preview PR/build] [DEBUG] Found revision: d6a347bf25f4ad74dc59b213c0b9d502eb14eb4e
[Preview PR/build] [DEBUG] Skipping local actions/checkout because workdir was already copied
[Preview PR/build] [DEBUG] skip pre step for 'Checkout code': no action model available
[Preview PR/build] [DEBUG] Found revision: d6a347bf25f4ad74dc59b213c0b9d502eb14eb4e
[Preview PR/build]   ☁  git clone 'https://github.com/docker/setup-buildx-action' # ref=v3
[Preview PR/build] [DEBUG]   cloning https://github.com/docker/setup-buildx-action to /Users/quasimodo/.cache/act/docker-setup-buildx-action@v3
[Preview PR/build] [DEBUG] Cloned https://github.com/docker/setup-buildx-action to /Users/quasimodo/.cache/act/docker-setup-buildx-action@v3
[Preview PR/build] [DEBUG] Checked out v3
[Preview PR/build] [DEBUG] Read action <nil> from 'Unknown'
[Preview PR/build] [DEBUG] skipping post step for 'publish pipeline artifact'; step was not executed
[Preview PR/build] [DEBUG] skipping post step for 'Build and push Docker image'; step was not executed
[Preview PR/build] [DEBUG] skipping post step for 'Login to DockerHub'; step was not executed
[Preview PR/build] [DEBUG] skipping post step for 'Set up Docker Buildx'; step was not executed
[Preview PR/build] [DEBUG] skipping post step for 'Checkout code'; step was not executed
[Preview PR/build] [DEBUG] Removed container: af91d489be091ec58aa912de0bcd31c1510619d0a1433ecb74feb66cd8d88bbe
[Preview PR/build] [DEBUG]   🐳  docker volume rm act-Preview-PR-build-d5e167f82ffe4fa1c7de90ba0b9f53096c9c177f9614a4bd6cc1fd8a9f19b9c5
[Preview PR/build] [DEBUG]   🐳  docker volume rm act-Preview-PR-build-d5e167f82ffe4fa1c7de90ba0b9f53096c9c177f9614a4bd6cc1fd8a9f19b9c5-env
[Preview PR/build] 🏁  Job succeeded
[Preview PR/build] [DEBUG] Found revision: d6a347bf25f4ad74dc59b213c0b9d502eb14eb4e
Error: The runs.using key in action.yml must be one of: [composite docker node12 node16], got node20
```

**Comment by Aya-Inocentes** [comment]
Hello, bumping this issue,

I'm using the actions/create-github-app-token@v1

**Comment by kpeters-cbsi** [comment]
This was fixed for me by updating to `act` 0.2.55

**Comment by liranbg** [comment]
> This was fixed for me by updating to `act` 0.2.55

confirming.

**Comment by darklight147** [comment]
Fixed on 0.2.55

**Comment by danielgoodwin97** [comment]
If anyone else runs into something like this, it turns out I had an ancient version of act which I had previously installed a long time ago. Try updating.

**Comment by aminya** [comment]
This fixed the issue for me:
```
gh extension upgrade gh-act
```

**Comment by maxiride** [comment]
Passing by since I just encountered the same error message. act installed through scoop and chocolatey fails with:
```
[docker-build-publish/docker]   ❌  Failure - Login to Docker Hub
Error: The runs.using key in action.yml must be one of: [composite docker node12], got node20
```
However, when running it as a github cli extension everything works fine. Since I got the job done I didn't investigate further, from what I experienced is _seems_ like not all act binaries are built equally.



---

My Dockerfile as of today.
```yml
name: docker-build-publish

on:
  push:
    tags:
      - 'v*.*.*'

jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      -
        name: Checkout
        uses: actions/checkout@v4
      -
        name: Login to Docker Hub
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
      -
        name: Docker meta
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: ${{ github.repository }}
      -
        name: Build and push
        uses: docker/build-push-action@v6
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
```

**Comment by sharq88** [comment]
> This was fixed for me by updating to `act` 0.2.55

I was getting the same despite a newer version: 

> Error: The runs.using key in action.yml must be one of: [composite docker node12 node16 node20], got node24

==> Upgrading act `0.2.74` -> `0.2.84` fixed it for me too.
